Add recipe for aws-athena-babel

will-abb opened the pull request:
### Brief summary of what the package does

`aws-athena-babel` allows you to run AWS Athena SQL queries directly from Org Babel source blocks in Emacs. It submits queries via the AWS CLI, monitors their execution in a live buffer, and provides options to display results as CSV or JSON. It also supports cost tracking and result reuse using Athena workgroups.

### Direct link to the package repository

https://github.com/will-abb/aws-athena-babel

### Your association with the package

I am the author and maintainer of this package.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

riscy commented:
Thanks for this.  Quick first pass -

⸺ `aws-athena-babel.el` with byte-compile using Emacs 30.1:
```
aws-athena-babel.el:100:15: Warning: reference to free variable `org-src-lang-modes'
aws-athena-babel.el:100:15: Warning: assignment to free variable `org-src-lang-modes'
In end of data:
aws-athena-babel.el:154:6: Warning: the function `persp-add-buffer' is not known to be defined.
```
Re: the `org` warnings, I think you need a `(require 'org)`.  To clean up the warning on `persp-add-buffer` you can use `declare-function` to do bookkeeping on this.

⸺ `aws-athena-babel.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- aws-athena-babel.el#L48: Use `(temporary-file-directory)` instead of /tmp in code
- aws-athena-babel.el#L428: Use `(temporary-file-directory)` instead of /tmp in code
- aws-athena-babel.el#L568: Use `(temporary-file-directory)` instead of /tmp in code
- aws-athena-babel.el#L613: Use `(temporary-file-directory)` instead of /tmp in code
```

`#'load`-check on each file:
```
Loading aws-athena-babel.el
  aws-athena-babel.el:Error: Emacs 30.1:
  (void-variable org-src-lang-modes)
```

⸺ Package and license:
- Prefer default recipe: `(aws-athena-babel :fetcher github :repo "will-abb/aws-athena-babel")`

will-abb commented:
Thanks for the detailed review — I’ve addressed all the points:

- [x] Added `(require 'org)` to handle the `org-src-lang-modes` warning  
- [x] Used `declare-function` for `persp-add-buffer` to resolve the byte-compiler warning  
- [x] Replaced all hardcoded `/tmp` references with `(temporary-file-directory)`  
- [x] Updated the recipe to use the preferred default without a `:files` entry  
- [x] Verified all changes using **melpazoid** — thanks for pointing me to it!

Let me know if there's anything else that needs to be changed.

riscy commented:
Thank you!  I think this is minor considering the user is the only one who will be changing these variables, but in `aws-athena-babel--build-start-query-command` I notice you should be using `shell-quote-argument` around the arguments, e.g. `(shell-quote-argument aws-athena-babel-query-file)`.  This'll protect users against accidentally having shell control characters or hanging spaces on these variables.

In case you haven't, ensure GitHub release v1.1.1 is up-to-date with your current code and `Package-Version`

No blockers, though.  Welcome to MELPA. :)
